From 98d6c9edb19a90935d51032692b240e7cc238834 Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Wed, 29 Mar 2017 12:26:15 +0200
Subject: [PATCH 03/11] MdeModulePkg/RamDiskDxe: fix C string literal
 catenation in info messages

Message-id: <20170329102623.11570-4-lersek@redhat.com>
Patchwork-id: 74570
O-Subject:  [RHEL-7.4 ovmf PATCH 03/11] MdeModulePkg/RamDiskDxe: fix C string
	literal catenation in info messages
Bugzilla: 1430262
Acked-by: Andrew Jones <drjones@redhat.com>
Acked-by: Al Stone <ahs3@redhat.com>
Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

RamDiskDxe installs the RamDiskAcpiCheck() Ready To Boot callback
function. If EFI_ACPI_TABLE_PROTOCOL and/or EFI_ACPI_SDT_PROTOCOL are not
found, then informational messages are logged, and the RAM disks are not
published to the (nonexistent) NFIT table.

The logic is fine, but the info messages are not concatenated correctly
from multiple string literals -- the second parts are passed as (unused)
arguments to DEBUG(). Fix the typos.

Cc: Ard Biesheuvel <ard.biesheuvel@linaro.org>
Cc: Feng Tian <feng.tian@intel.com>
Cc: Hao Wu <hao.a.wu@intel.com>
Cc: Leif Lindholm <leif.lindholm@linaro.org>
Cc: Star Zeng <star.zeng@intel.com>
Contributed-under: TianoCore Contribution Agreement 1.0
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
Reviewed-by: Feng Tian <feng.tian@intel.com>
Reviewed-by: Hao Wu <hao.a.wu@intel.com>
Reviewed-by: Star Zeng <star.zeng@intel.com>
(cherry picked from commit 76874be3d411bf8daac051718e20932e0bf97d70)
---
 MdeModulePkg/Universal/Disk/RamDiskDxe/RamDiskDriver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/MdeModulePkg/Universal/Disk/RamDiskDxe/RamDiskDriver.c b/MdeModulePkg/Universal/Disk/RamDiskDxe/RamDiskDriver.c
index d1dd13a..b2bafc5 100644
--- a/MdeModulePkg/Universal/Disk/RamDiskDxe/RamDiskDriver.c
+++ b/MdeModulePkg/Universal/Disk/RamDiskDxe/RamDiskDriver.c
@@ -74,7 +74,7 @@ RamDiskAcpiCheck (
   if (EFI_ERROR (Status)) {
     DEBUG ((
       EFI_D_INFO,
-      "RamDiskAcpiCheck: Cannot locate the EFI ACPI Table Protocol,",
+      "RamDiskAcpiCheck: Cannot locate the EFI ACPI Table Protocol, "
       "unable to publish RAM disks to NFIT.\n"
       ));
     return;
@@ -91,7 +91,7 @@ RamDiskAcpiCheck (
   if (EFI_ERROR (Status)) {
     DEBUG ((
       EFI_D_INFO,
-      "RamDiskAcpiCheck: Cannot locate the EFI ACPI Sdt Protocol,",
+      "RamDiskAcpiCheck: Cannot locate the EFI ACPI Sdt Protocol, "
       "unable to publish RAM disks to NFIT.\n"
       ));
     mAcpiTableProtocol = NULL;
-- 
1.8.3.1

